src/main/java/com/github/leroyguillaume/keycloak/bcrypt/BCryptPasswordHashProvider.java

```diff
@@ -37,7 +39,12 @@ public PasswordCredentialModel encodedCredential(final String rawPassword, final
     @Override
     public String encode(final String rawPassword, final int iterations) {
         final int cost = iterations == -1 ? defaultIterations : iterations;
-        return BCrypt.with(BCrypt.Version.VERSION_2A).hashToString(cost, rawPassword.toCharArray());
+        char[] pw = rawPassword.toCharArray();
+        if (pw.length > BCrypt.Version.MAX_PW_LENGTH_BYTE - 1) {
```

Comment:
In the case of encoding I think we should remove the condition and always use the same encoding regardless of the size of the password. Preferably an encoding that gives the user access to all 72 bytes. This will also serve as a passive migration for any current passwords that were already encoded with different version.

We probably want to just do `BCrypt.with(LongPasswordStrategies.truncate(BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR)).hashToString(cost, pw);` this appears to be the way it was done for another version of the library but anything that allows us to use the full 72 bytes should be ok.